src/crimson/net/SocketConnection.h

```diff
@@ -14,21 +14,16 @@
 
 #pragma once
 
-#include <seastar/core/gate.hh>
-#include <seastar/core/reactor.hh>
 #include <seastar/core/shared_future.hh>
 
-#include "msg/Policy.h"
+#include "Session.h"
 #include "Connection.h"
-#include "crimson/thread/Throttle.h"
-#include "Socket.h"
-
-class AuthSessionHandler;
 
 namespace ceph::net {
 
 class SocketConnection : public Connection {
-  std::optional<Socket> socket;
+  Session _s;
```

Comment:
why do we need to extract Session out of `SocketConnection` ?

Reply:
Please refer to the commit msg: https://github.com/ceph/ceph/pull/24142/commits/765316041546752adca6867e541b235e8d902f16

> * the `Session` class:
    async-msgr make `Protocol` a friend of `AsyncConnection` to access
    its internal states. This implementation extracts a `Session` struct
    to share access between `Protocol` and `SocketConnecton`.

But it is not final and may change in the future, because I think the next step is to enable concurrent lossy connections and the sharded seastar-msgr first. Then we need to reconsider how to deal with `Session` when enable lossless connections, it is also related to replacing logic IMO.